istio/datadog_checks/istio/istio.py

```diff
@@ -1,15 +1,41 @@
 # (C) Datadog, Inc. 2018-Present
 # All rights reserved
 # Licensed under Simplified BSD License (see LICENSE)
-from datadog_checks.base import OpenMetricsBaseCheck
 
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheck
+
+from .constants import BLACKLIST_LABELS
 from .legacy_1_4 import LegacyIstioCheck_1_4
+from .metrics import ISTIOD_METRICS
 
 
 class Istio(OpenMetricsBaseCheck):
 
     DEFAULT_METRIC_LIMIT = 0
 
+    def __init__(self, name, init_config, instances):
+        instance = instances[0]
+
+        if instance.get('istio_mesh_endpoint'):
+            raise ConfigurationError("istio_mesh_endpoint needs to be configured in a separate instance.")
```

Comment:
In typical k8s istio 1.5 environment, it is not a common use case to monitor both mesh metrics and istiod metrics in the same instance/annotation.  Documenting configuration error to properly warn against attempts to previous configuration. 

This new behavior would branch away from the previous multiple-endpoint instance type. IMO I think it's worth using the opportunity to navigate away from the previous behavior as it no longer applies to the new `1.5` architecture.

Reply:
```
if instance.get('istio_mesh_endpoint'):
            raise ConfigurationError("istio_mesh_endpoint needs to be configured in a separate instance.")
```
Seems I'm missing something really obvious here but wouldn't that same `ConfigurationError` be raised in the separate instance?